### PR TITLE
README updates

### PR DESCRIPTION
**Pull request by jamarshon**
We want to explain why we are choosing the conventions mentioned in https://github.com/pytorch/audio/pull/169.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,22 +72,25 @@ API Reference is located here: http://pytorch.org/audio/
...
+* `n_mel`, `n_mfcc`: the number of mel and MFCC bins
+* `n_freq`: the number of bins in a linear spectrogram
+* `min_freq`: the lowest frequency of the lowest band in a spectrogram
+* `max_freq`: the highest frequency of the highest band in a spectrogram
+* `win_length`: the length of the STFT window
+* `window_fn`: for functions that creates windows e.g. torch.hann_window
+
+Transforms expect the following dimensions.
 
 * Spectrogram: (channel, time) -> (channel, freq, time)
```
Comment (vincentqb) [inline README.md]: Can you also make the transformations use the `back tick`?

File: README.md [hunk 2]
```diff
@@ -63,22 +72,25 @@ API Reference is located here: http://pytorch.org/audio/
...
+* `hop_length`: the number of samples between the starts of consecutive frames
+* `n_fft`: the number of Fourier bins
+* `n_mel`, `n_mfcc`: the number of mel and MFCC bins
+* `n_freq`: the number of bins in a linear spectrogram
+* `min_freq`: the lowest frequency of the lowest band in a spectrogram
+* `max_freq`: the highest frequency of the highest band in a spectrogram
+* `win_length`: the length of the STFT window
+* `window_fn`: for functions that creates windows e.g. torch.hann_window
+
+Transforms expect the following dimensions.
```
Comment (vincentqb) [inline README.md]: Transformations or transforms?

File: README.md [hunk 3]
```diff
@@ -63,22 +72,25 @@ API Reference is located here: http://pytorch.org/audio/
...
-* win_length: the length of the STFT window
-* window_fn: for functions that creates windows e.g. torch.hann_window
-
-Transforms expect the following dimensions. In particular, the input of all transforms and functions assumes channel first.
+With torchaudio being a machine learning library and built on top of PyTorch,
+torchaudio is standardized around the following naming conventions. In particular,
+tensors are assumed to have channel as the first dimension and time as the last
+dimension (when applicable). This makes it consistent with PyTorch's dimensions.
+
+* `waveform`: a tensor of audio samples with dimensions (channel, time)
```
Comment (vincentqb) [inline README.md]: Do we want to provide explanations for the naming conventions, or just showing the examples is good enough?
* We use the conventions that channels are first.
* We call `n_*` variables referring to the number of `*`.
* etc

File: README.md (hunk 2 above)
Comment (jamarshon) [reply README.md]: Transforms I guess. To match the file name

File: README.md (hunk 3 above)
Comment (jamarshon) [reply README.md]: good point. done.

File: README.md [hunk 4]
```diff
@@ -63,28 +72,34 @@ API Reference is located here: http://pytorch.org/audio/
...
-* MFCC: (channel, time) -> (channel, mfcc, time)
-* MuLawEncode: (channel, time) -> (channel, time)
-* MuLawDecode: (channel, time) -> (channel, time)
-* Resample: (channel, time) -> (channel, time)
+With torchaudio being a machine learning library and built on top of PyTorch,
+torchaudio is standardized around the following naming conventions. In particular,
+tensors are assumed to have channel as the first dimension and time as the last
+dimension (when applicable). This makes it consistent with PyTorch's dimensions.
+For size names, the prefix `n_` is used (e.g. "a tensor of size (`n_freq`, `n_mel`)")
+whereas, dimension names do not have this prefix (e.g. "a tensor of
```
Comment (vincentqb) [inline README.md]: no coma after whereas

**Comment by jamarshon** [comment]
I think we mention formats supported already https://github.com/pytorch/audio/pull/180/files#diff-04c6e90faac2675aa89e2176d2eec7d8R15

File: README.md [hunk 5]
```diff
@@ -63,28 +72,47 @@ API Reference is located here: http://pytorch.org/audio/
...
-* MelScale: (channel, time) -> (channel, mel, time)
-* MelSpectrogram: (channel, time) -> (channel, mel, time)
-* MFCC: (channel, time) -> (channel, mfcc, time)
-* MuLawEncode: (channel, time) -> (channel, time)
-* MuLawDecode: (channel, time) -> (channel, time)
-* Resample: (channel, time) -> (channel, time)
+With torchaudio being a machine learning library and built on top of PyTorch,
+torchaudio is standardized around the following naming conventions. In particular,
+tensors are assumed to have channel as the first dimension and time as the last
+dimension (when applicable). This makes it consistent with PyTorch's dimensions.
```
Comment (vincentqb) [inline README.md]: why is it consistent with PyTorch?

**Comment by vincentqb** [comment]
> I think we mention formats supported already https://github.com/pytorch/audio/pull/180/files#diff-04c6e90faac2675aa89e2176d2eec7d8R15

If we intend to eventually remove sox, writing supported formats is a promise that we'll be breaking. IMHO we'll want to make sox optional, so that won't be a problem. In any case, maybe we should mention that the support comes from sox there?

**Pull request merged**